### ISSUE TITLE
i want to get the push notification in background

### ISSUE THREAD

**Issue by developuser** [phonegap/phonegap-plugin-push#2301]
### Expected Behaviour
i want to get the push notification in background or when the app is closed
### Actual Behaviour
push notification shows only the app in foreground
#### Platform and Version (eg. Android 5.0 or iOS 9.2.1)
android 6.0
#### (Android) What device vendor (e.g. Samsung, HTC, Sony...)
samsung
#### Cordova CLI version and cordova platform version
cordova android 7.1.0
    cordova --version                                    # e.g. 6.0.0
    cordova platform version android                     # e.g. 4.1.1

#### Plugin version
phonegap-plugin-push 2.2.2
    cordova plugin version | grep phonegap-plugin-push   # e.g. 1.5.3

#### Logs taken while reproducing problem
using this plugin i just get the push notification message and alerted it but i cant know how to display it as a notification also in back ground

**Comment by macdonst** [comment]
@developuser I don't actually understand your question. When the app is in the background or closed an incoming notification should be displayed in the notification shade. If the app is in the foreground the `on('notification')` event listener will be called. 

If the app is in the background and you want the `on('notification')` event listener to be called you would use [Background Notifications](https://github.com/phonegap/phonegap-plugin-push/blob/master/docs/PAYLOAD.md#background-notifications).

**Comment by developuser** [comment]
@macdonst May i know this rerence code as am beginer to a ionic i cant know about it

**Comment by macdonst** [comment]
@developuser you shouldn't need to do anything with the push notification for it to show up in the shade in the background. The plugin should take care of adding it to the shade.

**Comment by matthijsbierman** [comment]
We ran into the same problem today, and found that on Android 8 / Oreo the notification no longer displays if you are sending it with only a `data` payload.
You need to include a `notification` payload, e.g.:

`"notification": {
	"title":"App",
	"body":"Message text here."
}`

On Android 7 and below it does work if you send a `data` payload with `title` and `message`. All of the current documentation around GCM/FCM shows that you should send a `notification` payload, but I'm quite sure that this was not always the case. E.g. the official [GCM server](https://github.com/google/gcm) Java library, which hasn't been updated in a while, does not support adding anything but data payload in version 1.0.0.

TLDR; use the `notification` payload instead of only `data`

**Comment by matthijsbierman** [comment]
Handling a clicked push notification which was received in the background apparently needs to be done through an Intent. The documentation is sparse:
https://firebase.google.com/docs/cloud-messaging/android/receive

**Comment by macdonst** [comment]
Matthijs I'm away from my laptop right now but I'll reply later as you've
made an erroneous assumption that may lead you to more problems down the
road.

**Comment by matthijsbierman** [comment]
@macdonst thanks for the quick reply! I'll wait for you to avoid wasting my time.

**Comment by macdonst** [comment]
@matthijsbierman okay, so if you put the `notification` section in your push payload you will run into the problem [documented here](https://github.com/phonegap/phonegap-plugin-push/blob/master/docs/PAYLOAD.md#notification-vs-data-payloads). The short version is when the OS sees the `notification` part of the push payload it never passes the push notification along to the plugin, instead it handles it at the OS level. This means many of the more advanced features of the plugin will not work.

The reason why pushes on Android 8 stop working with version 1.10.7 of this plugin is that support for Android 8 didn't land until version 2.1.0 of this plugin. Android 8 or Oreo requires all apps to setup [NotificationChannel](https://developer.android.com/guide/topics/ui/notifiers/notifications.html#ManageChannels). So if your app is targeting API 26 or higher you should use v2.x of the PushPlugin and FCM.

**Comment by matthijsbierman** [comment]
@macdonst thanks for taking the time to respond.

The app is targeted at API level 19 (Android 4.4 / KitKat), using cordova-android 6.3.0. It appears that the plugin works fine, it registers for a token and it receives the push messages in the foreground. In the background `adb logcat` still shows that something is coming in, but the notification is simply not displayed.

The problem appears to be that on phones running Android 8, the trick described in _Notification vs Data Payloads_ no longer works. The payload:

```{
    "data" : {
        "title": "Test Notification",
        "body": "This offer expires at 11:30 or whatever",
        "notId": 10,
        "surveyID": "ewtawgreg-gragrag-rgarhthgbad"
    }
}
```

does not result in a push notification from the OS in the top bar any longer. The same push message, to the same app works on Android 7.

I think that's what more people here are experiencing.

As far as the problem with the `notification` payload not triggering the plugin's `onMessageReceived` method - it looks like that is [as designed](https://firebase.google.com/docs/cloud-messaging/android/receive):

> **onMessageReceived** is provided for most message types, with the following exceptions:
>
> - **Notification messages delivered when your app is in the background.** In this case, the notification is delivered to the device’s system tray. A user tap on a notification opens the app launcher by default.
> - **Messages with both notification and data payload, both background and foreground.** In this case, the notification is delivered to the device’s system tray, and the data payload is delivered in the extras of the intent of your launcher Activity.

**Comment by matthijsbierman** [comment]
Found an error in logcat which indicates you are probably right about the notification not showing due to the missing notification channel.

> E NotificationService: No Channel found for pkg=com.x.x, channelId=null, id=0, tag=x, opPkg=com.x.x, callingUid=10292, userId=0, incomingUserId=0, notificationUid=10292, notification=Notification(channel=null pri=0 contentView=null vibrate=default sound=content://settings/system/notification_sound tick defaults=0x2 flags=0x10 color=0x00000000 number=0 vis=PRIVATE semFlags=0x0 semPriority=0 semMissedCount=0)

Will try upgrading the plugin to 2.x branch and switching to FCM. Thanks for your help!

**Comment by matthijsbierman** [comment]
App was not targeting API level 19. There was no `targetSkdVersion` defined and therefore targeting level 26. Will try setting the target API level to 25 to see if the compatibility options fix the problem.

**Comment by developuser** [comment]
@macdonst  i had tried but i get the same errror my plugin versioniis 2.2.2, and the notification handling code is    



$rootScope.$on('$cordovaPushV5:notificationReceived', function(event, data) {  // use two variables here, event and data !!!
    if (data.additionalData.foreground === false) {
      
      console.log(data.message);
    alert(data.message);
        // do something if the app is in foreground while receiving to push - handle in app push handling
    } else {
        console.log(data.message);
        alert(data.message);
       // handle push messages while app is in background or not started
    }  
    $cordovaPushV5.finish().then(function (result) {
        // OK finished - works only with the dev-next version of pushV5.js in ngCordova as of February 8, 2016
    }, function (err) {
        // handle error
    });
});

$rootScope.$on('$cordovaPushV5:errorOccurred', function(event, error) {
    // handle error
});
and the server payload is 


{"registration_ids":["e3xcIMg0kq8:APA91bGozaNTx6t429d_bWt"],"data":{"title":"test","message":"test"}}


as am beginner to this icant able to find the solution

**Comment by macdonst** [comment]
@developuser I need clarification from you. When you force close the app and you send a push message do you see the notification in the notification shade?

**Comment by maitscha** [comment]
I have the same problem ... regarding "the data payload is delivered in the extras of the intent of your launcher Activity", is this already somehow handled from this plugin when a background Push notification is received and the Cordova app is opened?
